src/chains/ethereum/ethereum/tests/api/eth/createAccessList.test.ts

```diff
@@ -0,0 +1,474 @@
+import assert from "assert";
+import { EthereumProvider } from "../../../src/provider";
+import getProvider from "../../helpers/getProvider";
+import compile, { CompileOutput } from "../../helpers/compile";
+import { join } from "path";
+import { BUFFER_EMPTY, Data, Quantity } from "@ganache/utils";
+import { Ethereum } from "../../../src/api-types";
+import { AccessList } from "@ganache/ethereum-transaction/src/access-lists";
+import { Address } from "@ganache/ethereum-address";
+import { CallError } from "@ganache/ethereum-utils";
+import { ERROR, VmError } from "@ethereumjs/vm/dist/exceptions";
+import { EVMResult } from "@ethereumjs/vm/dist/evm/evm";
+
+const encodeValue = (val: string) => {
+  return Data.toString(val, 32).slice(2);
+};
+
+async function deployContract(provider, from, code) {
+  await provider.send("eth_subscribe", ["newHeads"]);
+  const transactionHash = await provider.send("eth_sendTransaction", [
+    {
+      from,
+      data: code,
+      gas: "0xfffff"
+    }
+  ]);
+  await provider.once("message");
```

Comment:
Since eth_subscribe/await message stuff is not needed since we default to eager mode should we keep doing it in our tests? :thinking: I'm torn because it is technically correct... buuuuttt the way we do it isn't 100% correct (since maybe something else also subscribed to a some different thing, and then the `once("message")` might not be the message we were waiting for)

In any case... this function should also `unsubscribe` at the end.

Reply:
Added an unsubscribe and opened an issue! https://github.com/trufflesuite/ganache/issues/3754